src/todos/todos.service.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { Test, TestingModule } from '@nestjs/testing'
+import { TodosService } from './todos.service'
+import { TodosDatabaseService } from './todos.database.service'
+import type { Todo } from './schemas/todos.schema'
+import { createMock } from '@golevelup/ts-jest'
+
+describe('TodosService', () => {
+  let todosService: TodosService
+  let todosDatabaseService: TodosDatabaseService
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        TodosService,
+        {
+          provide: TodosDatabaseService,
+          useValue: createMock<TodosDatabaseService>(),
+        },
+      ],
+    }).compile()
+
+    todosService = module.get<TodosService>(TodosService)
+    todosDatabaseService =
+      module.get<TodosDatabaseService>(TodosDatabaseService)
+  })
+
+  it('should call create method with correct arguments', async () => {
+    const todoItem: Omit<Todo, '_id'> = {
+      title: 'Test title',
+      description: 'Test description',
+    }
+    await todosService.create(todoItem as Todo)
+
+    expect(todosDatabaseService.create).toHaveBeenCalledWith(todoItem)
+  })
+
+  it('should call getAll method with correct arguments', async () => {
+    await todosService.getAll()
+
+    expect(todosDatabaseService.getAll).toHaveBeenCalled()
+  })
+
+  it('should call getById method with correct arguments', async () => {
+    const id = '1'
+    await todosService.getById(id)
+
+    expect(todosDatabaseService.getById).toHaveBeenCalledWith(id)
+  })
+
+  it('should call update method with correct arguments', async () => {
+    const id = '1'
+    const newParams: Partial<Todo> = { description: 'test updated description' }
```

Comment:
Shouldn't we use UpdateTodoDto as type here?

Reply:
Yep, fixed